File: src/python/engine/quiz.py
from datetime import datetime, timedelta

srs_map = (
    timedelta(hours=4),
    timedelta(hours=8),
    timedelta(days=1),
    timedelta(days=3),
    timedelta(weeks=1),
    timedelta(weeks=2),
    timedelta(weeks=4),
    timedelta(weeks=16)
)


def get_next_review(srs_level: int) -> datetime:
    try:
        return srs_map[srs_level - 1] + datetime.now()
    except IndexError:
        return repeat_review()


def repeat_review() -> datetime:
    return datetime.now() + timedelta(minutes=10)

File: src/node/engine/quiz.ts
import moment from "moment";

export const srsMap = [
    moment.duration(4, "hour"),
    moment.duration(8, "hour"),
    moment.duration(1, "day"),
    moment.duration(3, "day"),
    moment.duration(1, "week"),
    moment.duration(2, "week"),
    moment.duration(4, "week"),
    moment.duration(16, "week")
];

export function getNextReview(srsLevel: number): Date {
    let toAdd = srsMap[srsLevel];
    toAdd = toAdd === undefined ? moment.duration(10, "minute") : toAdd;
    return moment().add(toAdd).toDate();
}

export function repeatReview(): Date {
    return moment().add(10, "minute").toDate();
}
